board_test.go

```diff
@@ -0,0 +1,73 @@
+package honeycombio
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestBoards(t *testing.T) {
+	var b *Board
+	var err error
+
+	c := newTestClient(t)
+
+	t.Run("Create", func(t *testing.T) {
+
+		data := &Board{
+			Name:        "Test Board",
+			Description: "A board with some queries",
+			Style:       "visual",
+			Queries: []BoardQuery{
+				{
+					Caption: "A sample dataset",
+					Dataset: c.dataset,
+					Query:   QuerySpec{},
+				},
+			},
+		}
+		b, err = c.Boards.Create(data)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		assert.NotNil(t, b.ID)
+		assert.Equal(t, data.Name, b.Name)
+		assert.Equal(t, data.Description, b.Description)
+		assert.Equal(t, data.Style, b.Style)
+		assert.Equal(t, data.Queries, b.Queries)
+	})
+
+	t.Run("List", func(t *testing.T) {
+		boards, err := c.Boards.List()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		var createdBoard *Board
+
+		for _, board := range boards {
+			if board.ID == b.ID {
+				createdBoard = &board
+			}
```

Comment:
```suggestion
			if board.ID == b.ID {
				createdBoard = &board
				break
			}
```
For some reason, the test only passes on my system if add a `break` here but I don't get why that would happen 🤔 
Without `break` the `createdBoard` ends up being a totally unrelated board.

Reply:
Weird, I wonder what's going on there.  

This was passing on a new test account for me (and the trigger test was failing).

I'd say let's just add the `break` and investigate any additional weirdness as it comes up in the tf provider